### ISSUE TITLE
weird render: image matrix seems not fit well

### ISSUE THREAD

**Issue by cocodrino** [PaulLeCam/react-leaflet#91]
Hi, my map is render in a very stranger way, seems image location doesnt correspon with its neighborhood images (and some images doesnt even render), I'm not sure how explain this so I'll show you an image

![screenshot from 2015-11-04 16 45 26](https://cloud.githubusercontent.com/assets/616412/10952207/cb4ceeb2-8313-11e5-92ed-4491fad74217.png)
 
the code is 

```js
import React, { Component } from 'react';
var Map = require('react-leaflet').Map;
var Marker = require('react-leaflet').Marker;
var Popup = require('react-leaflet').Popup;
var Tile = require('react-leaflet').TileLayer;


const place = ['10.5', '-66.9167'];
export default class App extends Component {
  render() {
    return (
      <Map
        center= {place} zoom={13}
        >

        <Tile
          url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          ></Tile>

        <Marker position={place}>


        </Marker>

      </Map>
    );
  }
}

```

thank so much!!!!

**Comment by cocodrino** [comment]
update: I've tried the same code in the doc

```js
import React from 'react';
import ReactDOM from 'react-dom';
 

import { Map, Marker, Popup, TileLayer } from 'react-leaflet';

const position = [51.505, -0.09];
const map = (
  <Map center={position} zoom={13}>
    <TileLayer
      url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
      attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
      />
    <Marker position={position}>
      <Popup>
        <span>A pretty CSS3 popup.<br/>Easily customizable.</span>
      </Popup>
    </Marker>
  </Map>
);


ReactDOM.render(map, document.getElementById('root'));

```

with identical results, the image matrix is not correct..thanks!!

**Comment by AngusMackHPE** [comment]
What's your HTML and CSS look like?

You need a bounding div with dimensions to insert the map into, otherwise you see this sort of behaviour.

**Comment by PaulLeCam** [comment]
Hi,
It looks like you are not including Leaflet's CSS, see the "Preparing your page" section in [Leaflet's quick start guide](http://leafletjs.com/examples/quick-start.html).

**Comment by cocodrino** [comment]
ups..sorry @PaulLeCam...thank so much!!!

**Comment by tormod17** [comment]
We have the same problem, it would be great to see the code you used to get past this.

**Comment by ashley-christensen** [comment]
I solved this bug by doing the following and it worked great!

-Adding INLINE styling to the MapContainer component Directly
-Adding an ACTUAL height to this Component [in pixels or otherwise]
-Importing the StyleSheet -- import "leaflet/dist/leaflet.css";

Others were having issues trying to add styling to the div, as per the documentation. This did not work for me until I tried:

< MapContainer
style={{ width: '100%', height: '600px' }}
center={position}
zoom={13} >

Hope this helps!
